tools/driver/p4c_src/main.py

```diff
@@ -136,69 +154,134 @@ def main():
-                            "invocations of the same subparser instance.",
+                        help="Enable optimization of inlining of callee "
+                             "parsers (subparsers). "
+                             "The optimization is disabled by default. "
+                             "When the optimization is disabled, "
+                             "for each invocation of the subparser "
+                             "all states of the subparser are inlined, "
+                             "which means that the subparser "
+                             "might be inlined multiple times even if "
+                             "it is the same instance "
+                             "which is invoked multiple times. "
+                             "When the optimization is enabled, the compiler "
+                             "tries to identify the cases where it can inline"
+                             " the subparser's states only once for multiple "
+                             "invocations of the same subparser instance.",
                         action="store_true", default=False)
 
-    if (os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"):
+    ### DRYified “env_indicates_developer_build”
+    env_indicates_developer_build = os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"
+    if env_indicates_developer_build:
         add_developer_options(parser)
 
-    parser.add_argument("source_file", nargs='?', help="Files to compile", default=None)
+    parser.add_argument("P4_source_files", nargs='*',
+                        help="P4 source files to compile.", default=None)
 
     # load supported configuration.
     # We load these before we parse options, so that backends can register
     # proprietary options
     cfg_files = glob.glob("{}/*.cfg".format(os.environ['P4C_CFG_PATH']))
-    cfg = config.Config(config_prefix = "p4c")
+    cfg = config.Config(config_prefix="p4c")
     for cf in cfg_files:
         cfg.load_from_config(cf, parser)
+    ### ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    ###
+    ### Abe`s note: I hope my {“source_file” → “P4_source_files”} change
+    ###   (to the “parser” object) at this point in time can`t cause any
+    ###   ill effects in the callee “config.Config.load_from_config”.
+    ###
+    ###   WIP To Do: check with somebody more experienced with this codebase
+    ###              before even _thinking_ about merging
+    ###              this changeset into “main”.
 
     # parse the arguments
     opts = parser.parse_args()
 
+
+
     user_defined_version = os.environ.get('P4C_DEFAULT_VERSION')
-    if user_defined_version != None:
+    ### Note: the next line of real code _was_τhe clear and easy-to-understand
+    ###       “if user_defined_version != None:”, but “pycodestyle”
+    ###       complained about that idiom so I rewrote it to something
+    ###       semantically-equivalent but with different syntax.
+    if not (user_defined_version is None):
```

Comment:
I think that the python style is "is not None"

Reply:
Yes, but I have observed a bug in the Python language: “is not (None)” and “is (not None)” mean completely different things, and IMO that is a language-design bug, so I am just avoiding that ambiguity/problem on purpose by writing the code in a way that is both unambiguous and obviously means what it really means.  I hope that is OK.  😉

I guess, if we were designing a Python-like language from scratch, and we didn`t mind non-letter characters in key“words”, we could create an “isn`t” operator...  but that is probably just yet another of my silly programming dreams. 😉

Reply:
Oh, and “is_not” would work fine as a key“word” too, of course. 😉